### ISSUE TITLE
Error while installing app through Rancher CLI

### ISSUE THREAD

**Issue by rishabhmsra** [rancher/rancher#36332]
**Rancher Server Setup**
- Rancher version: v2.6.3
- Installation option (Docker install/Helm Chart): Docker install
 
**Information about the Cluster**
- Kubernetes version: v1.21.8-rancher2-1 `(4 nodes -> 1-cp, 1-etcd, 2-worker)`
- Cluster Type (Local/Downstream): Downstream aws node driver cluster
   
**Describe the bug**
- While installing the app using `rancher cli v2.4.14` getting panic error.

**To Reproduce**
- On rancher v2.6.3, provision a downstream cluster.
- Using `rancher cli v2.4.14` login into rancher server : `rancherctl login <rancher-server-url> --skip-verify -t <admin-token>`
- After running above command, select the `Default` project in the downstream cluster.
- Run the command : `rancherctl app install redis appFoo`
- output : 
```
$ rancherctl app install redis appFoo
panic: runtime error: index out of range [-1]

goroutine 1 [running]:
github.com/rancher/cli/cmd.getTemplateLatestVersion(0xc000797200, 0xc000232940, 0xc0003003a0, 0x20, 0xc000797200)
	/go/src/github.com/rancher/cli/cmd/app.go:1030 +0xaf
github.com/rancher/cli/cmd.templateInstall(0xc000256000, 0xc000186400, 0xc000186480)
	/go/src/github.com/rancher/cli/cmd/app.go:669 +0x6c5
github.com/urfave/cli.HandleAction(0xb0f600, 0xc7a6c8, 0xc000256000, 0xc000256000, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:524 +0x105
github.com/urfave/cli.Command.Run(0xc3fddb, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc4b138, 0x17, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:173 +0x579
github.com/urfave/cli.(*App).RunAsSubcommand(0xc0001b6fc0, 0xc00016dce0, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:405 +0x914
github.com/urfave/cli.Command.startApp(0xc3d126, 0x4, 0x0, 0x0, 0xc0001f1580, 0x1, 0x1, 0xc686c6, 0x74, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:372 +0x7ff
github.com/urfave/cli.Command.Run(0xc3d126, 0x4, 0x0, 0x0, 0xc0001f1580, 0x1, 0x1, 0xc686c6, 0x74, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:102 +0x9d4
github.com/urfave/cli.(*App).Run(0xc0001b6e00, 0xc000180d00, 0x5, 0x8, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:277 +0x808
main.mainErr(0xc000114058, 0x41aa01)
	/go/src/github.com/rancher/cli/main.go:124 +0x20cf
main.main()
	/go/src/github.com/rancher/cli/main.go:57 +0x25
```

**Comment by samjustus** [comment]
cli 2.4.14 doesn't work with rancher 2.6.3, should use 2.6.0 instead

**Comment by rishabhmsra** [comment]
Re-opening the issue.

Tried installing app on `rancher v2.6.4` using `CLI v2.6.4` and getting the panic error:
- On rancher v2.6.4, created downstream cluster `v1.22.7-rancher1-2` custom cluster (1-cp, 1-etcd, 3-worker)

```
$ rancherctl --version
rancher version v2.6.4

$ rancherctl apps install cattle-global-data:helm3-library-openebs --no-prompt -n default --version 3.0.0
panic: runtime error: index out of range [-1]

goroutine 1 [running]:
github.com/rancher/cli/cmd.getTemplateLatestVersion(0xc0006f8900, 0xc0002c5230, 0xc0006bc000, 0x28, 0xc0006f8900)
	/go/src/github.com/rancher/cli/cmd/app.go:1042 +0xaf
github.com/rancher/cli/cmd.templateInstall(0xc0002b4840, 0xc0002aad00, 0xc0002aae10)
	/go/src/github.com/rancher/cli/cmd/app.go:680 +0x6f7
github.com/urfave/cli.HandleAction(0xb33420, 0xcae4b8, 0xc0002b4840, 0xc0002b4840, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:524 +0x105
github.com/urfave/cli.Command.Run(0xc72f12, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc7e4ce, 0x17, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:173 +0x579
github.com/urfave/cli.(*App).RunAsSubcommand(0xc000216fc0, 0xc0002b4580, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:405 +0x914
github.com/urfave/cli.Command.startApp(0xc70223, 0x4, 0x0, 0x0, 0xc00024d5d0, 0x1, 0x1, 0xc9bd09, 0x74, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:372 +0x7ff
github.com/urfave/cli.Command.Run(0xc70223, 0x4, 0x0, 0x0, 0xc00024d5d0, 0x1, 0x1, 0xc9bd09, 0x74, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:102 +0x9d4
github.com/urfave/cli.(*App).Run(0xc000216e00, 0xc0000e6400, 0x9, 0x10, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:277 +0x808
main.mainErr(0xc000036178, 0x41aa01)
	/go/src/github.com/rancher/cli/main.go:124 +0x214f
main.main()
	/go/src/github.com/rancher/cli/main.go:56 +0x25
```

**Comment by cbron** [comment]
@rishabhmsra where is the 3.0.0 chart coming from ? I don't see that in our repo: https://github.com/rancher/helm3-charts/tree/master/charts/openebs
Do you still get the error when using version 2.5.0 ?

**Comment by rishabhmsra** [comment]
Hey @cbron, I tried with 2.5.0 as well but getting same error. Also i tried installing redis but got same error.

- Tested on rancher v2.6.4(docker install), provisioned single node, k8s v1.22.7-rancher1-2, ec2 node driver cluster :
```
rancherctl apps install cattle-global-data:helm3-library-openebs --no-prompt -n default --version 2.5.0
panic: runtime error: index out of range [-1]

goroutine 1 [running]:
github.com/rancher/cli/cmd.getTemplateLatestVersion(0xc00064a900, 0xc000245230, 0xc000b2bd10, 0x28, 0xc00064a900)
	/go/src/github.com/rancher/cli/cmd/app.go:1042 +0xaf
github.com/rancher/cli/cmd.templateInstall(0xc000232b00, 0xc00022af00, 0xc00022b050)
	/go/src/github.com/rancher/cli/cmd/app.go:680 +0x6f7
github.com/urfave/cli.HandleAction(0xb33420, 0xcae4b8, 0xc000232b00, 0xc000232b00, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:524 +0x105
github.com/urfave/cli.Command.Run(0xc72f12, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc7e4ce, 0x17, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:173 +0x579
github.com/urfave/cli.(*App).RunAsSubcommand(0xc000196fc0, 0xc000232840, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:405 +0x914
github.com/urfave/cli.Command.startApp(0xc70223, 0x4, 0x0, 0x0, 0xc0001cd5d0, 0x1, 0x1, 0xc9bd09, 0x74, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:372 +0x7ff
github.com/urfave/cli.Command.Run(0xc70223, 0x4, 0x0, 0x0, 0xc0001cd5d0, 0x1, 0x1, 0xc9bd09, 0x74, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:102 +0x9d4
github.com/urfave/cli.(*App).Run(0xc000196e00, 0xc0000e6400, 0x9, 0x10, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:277 +0x808
main.mainErr(0xc000036178, 0x41aa01)
	/go/src/github.com/rancher/cli/main.go:124 +0x214f
main.main()
	/go/src/github.com/rancher/cli/main.go:56 +0x25
```

- Tried installing redis using below command:
```
rancherctl app install redis appFoo
panic: runtime error: index out of range [-1]

goroutine 1 [running]:
github.com/rancher/cli/cmd.getTemplateLatestVersion(0xc000d38900, 0xc0002d51a0, 0xc000180080, 0x20, 0xc000d38900)
	/go/src/github.com/rancher/cli/cmd/app.go:1042 +0xaf
github.com/rancher/cli/cmd.templateInstall(0xc0002be9a0, 0xc0002aef00, 0xc0002af050)
	/go/src/github.com/rancher/cli/cmd/app.go:680 +0x6f7
github.com/urfave/cli.HandleAction(0xb33420, 0xcae4b8, 0xc0002be9a0, 0xc0002be9a0, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:524 +0x105
github.com/urfave/cli.Command.Run(0xc72f12, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc7e4ce, 0x17, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:173 +0x579
github.com/urfave/cli.(*App).RunAsSubcommand(0xc00021efc0, 0xc0002be6e0, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:405 +0x914
github.com/urfave/cli.Command.startApp(0xc70223, 0x4, 0x0, 0x0, 0xc00025d410, 0x1, 0x1, 0xc9bd09, 0x74, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:372 +0x7ff
github.com/urfave/cli.Command.Run(0xc70223, 0x4, 0x0, 0x0, 0xc00025d410, 0x1, 0x1, 0xc9bd09, 0x74, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:102 +0x9d4
github.com/urfave/cli.(*App).Run(0xc00021ee00, 0xc0001e8c80, 0x5, 0x8, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:277 +0x808
main.mainErr(0xc000114058, 0x41aa01)
	/go/src/github.com/rancher/cli/main.go:124 +0x214f
main.main()
	/go/src/github.com/rancher/cli/main.go:56 +0x25
```

**Comment by maxsokolovsky** [comment]
Reproduced this using Rancher `v2.6.4` and the latest CLI version.
Looks like the CLI requests the template asking to also drop versions incompatible with the rancher server version.
This results in not having any versions at all (for some charts, apparently), which we don't anticipate (but should), which results in a panic.

**Comment by cbron** [comment]
Fixing panic would be first step here then. Not sure about rest of it.

**Comment by MKlimuszka** [comment]
The chart tested for this isn't available for use in this version, so I am removing the blocker label

**Comment by rishabhmsra** [comment]
Re-opening this issue:

On rancher v2.6.5, provisioned single node, ec2 node driver cluster(k8s v1.23.6-rancher1-1). Getting panic error using CLI `v2.6.5`:

```
rancherctl apps install cattle-global-data:helm3-library-openebs --no-prompt -n default --version 2.5.0
panic: runtime error: index out of range [-1]

goroutine 1 [running]:
github.com/rancher/cli/cmd.getTemplateLatestVersion(0xc000e3e900, 0xc000245230, 0xc000930000, 0x28, 0xc000e3e900)
	/go/src/github.com/rancher/cli/cmd/app.go:1042 +0xaf
github.com/rancher/cli/cmd.templateInstall(0xc000236580, 0xc000220f00, 0xc000221050)
	/go/src/github.com/rancher/cli/cmd/app.go:680 +0x6f7
github.com/urfave/cli.HandleAction(0xb33420, 0xcae4b8, 0xc000236580, 0xc000236580, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:524 +0x105
github.com/urfave/cli.Command.Run(0xc72f12, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc7e4ce, 0x17, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:173 +0x579
github.com/urfave/cli.(*App).RunAsSubcommand(0xc000196fc0, 0xc0002362c0, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:405 +0x914
github.com/urfave/cli.Command.startApp(0xc70223, 0x4, 0x0, 0x0, 0xc0001cf5d0, 0x1, 0x1, 0xc9bd03, 0x74, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:372 +0x7ff
github.com/urfave/cli.Command.Run(0xc70223, 0x4, 0x0, 0x0, 0xc0001cf5d0, 0x1, 0x1, 0xc9bd03, 0x74, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:102 +0x9d4
github.com/urfave/cli.(*App).Run(0xc000196e00, 0xc0000e8400, 0x9, 0x10, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:277 +0x808
main.mainErr(0xc000036178, 0x41aa01)
	/go/src/github.com/rancher/cli/main.go:124 +0x214f
main.main()
	/go/src/github.com/rancher/cli/main.go:56 +0x25
```
-----------------------------------------

```
rancherctl app install redis appFoo
panic: runtime error: index out of range [-1]

goroutine 1 [running]:
github.com/rancher/cli/cmd.getTemplateLatestVersion(0xc000545200, 0xc00035b1a0, 0xc0007dc0a0, 0x20, 0xc000545200)
	/go/src/github.com/rancher/cli/cmd/app.go:1042 +0xaf
github.com/rancher/cli/cmd.templateInstall(0xc00037e2c0, 0xc000331200, 0xc000331290)
	/go/src/github.com/rancher/cli/cmd/app.go:680 +0x6f7
github.com/urfave/cli.HandleAction(0xb33420, 0xcae4b8, 0xc00037e2c0, 0xc00037e2c0, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:524 +0x105
github.com/urfave/cli.Command.Run(0xc72f12, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc7e4ce, 0x17, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:173 +0x579
github.com/urfave/cli.(*App).RunAsSubcommand(0xc0002a6fc0, 0xc00037e000, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:405 +0x914
github.com/urfave/cli.Command.startApp(0xc70223, 0x4, 0x0, 0x0, 0xc0002e5410, 0x1, 0x1, 0xc9bd03, 0x74, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:372 +0x7ff
github.com/urfave/cli.Command.Run(0xc70223, 0x4, 0x0, 0x0, 0xc0002e5410, 0x1, 0x1, 0xc9bd03, 0x74, 0x0, ...)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/command.go:102 +0x9d4
github.com/urfave/cli.(*App).Run(0xc0002a6e00, 0xc000270c80, 0x5, 0x8, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli@v1.22.5/app.go:277 +0x808
main.mainErr(0xc000180058, 0x41aa01)
	/go/src/github.com/rancher/cli/main.go:124 +0x214f
main.main()
	/go/src/github.com/rancher/cli/main.go:56 +0x25
```

**Comment by sowmyav27** [comment]
@brandonsuse Can you please let know what chart can be used going forward?

**Comment by PennyScissors** [comment]
@sowmyav27 @rishabhmsra All charts in catalog v1 are hidden, and cli does not support catalog v2. There is no existing chart that can be used in the cli tests except Longhorn which they are working on migrating and deprecating as well. At this point we need to consider deprecating cli or rewriting it to support catalog v2.

**Comment by brandonsuse** [comment]
@sowmyav27 I do not know the answer to this.

**Comment by AndrewHoffmanQA** [comment]
@nickwsuse and I tag teamed this bug. 

**Pass** (see pic) verified in `2.6-head Commit ID c8d01c6`

1. Used rancher cli v2.4.14 logged into rancher server `: rancherctl login <rancher-server-url> --skip-verify -t <admin-token>`
2. After running above command, selected the Default project in the downstream cluster.
3. Ran the command : `rancherctl app install redis appFoo`

**result**
see screenshot. bug has been fixed in the 2.6 release.

![image.png](https://zube.io/files/rancher/623e35b15eec51d4cdcf39c407bd489a-image.png)
